models/POP/shell_scripts/cesm2_1/CESM2_1_setup_pmo

```diff
@@ -0,0 +1,763 @@
+
+if ( $CASEROOT == `dirname $0` ) then
+   echo "ERROR: the setup script should not be located in the CASEROOT"
+   echo "directory, because all files in the CASEROOT dir will be removed"
+   echo "before creating the new case.  move the script to a safer place."
+   exit -1
+endif
+
+echo "removing old files from ${CASEROOT}"
+echo "removing old files from ${EXEROOT}"
+echo "removing old files from ${RUNDIR}"
+${REMOVE} ${CASEROOT}
+${REMOVE} ${EXEROOT}
+${REMOVE} ${RUNDIR}
+
+# ==============================================================================
+# Invoking create_newcase.
+# The --run-unsupported option is necessary for CESM2 because this compset
+# hasn't been tested.
+# ==============================================================================
+
+set use_tasks_per_node = 36
+set nthreads = 1
+
+echo ''
+echo 'Invoking create_newcase.'
+echo ''
+echo '${cesmroot}/cime/scripts/create_newcase'
+echo '--case ${CASEROOT}'
+echo '--mach ${mach}'
+echo '--res ${resolution}'
+echo '--compset ${compset}'
+echo '--run-unsupported'
+echo '--pecount ${use_tasks_per_node}x${nthreads}'
+
+${cesmroot}/cime/scripts/create_newcase \
+--case ${CASEROOT} \
+--mach ${mach} \
+--res ${resolution} \
+--compset ${compset} \
+--run-unsupported \
+--pecount ${use_tasks_per_node}x${nthreads}
+
+if ( $status != 0 ) then
+   echo "ERROR: Case could not be created."
+   exit -1
+endif
+
+# preserve a copy of this script as it was run
+set ThisFileName = $0:t
+${COPY} $ThisFileName ${CASEROOT}/${ThisFileName}.original
+${COPY} DART_params.csh ${CASEROOT}
+
+# ==============================================================================
+# Record the DARTROOT directory and copy the DART setup script to CASEROOT.
+# CESM_DART_config can be run at some later date if desired, but it presumes
+# to be run from a CASEROOT directory. If CESM_DART_config does not exist locally,
+# then it better exist in the expected part of the DARTROOT tree.
+# ==============================================================================
+
+if ( ! -e CESM_DART_config ) then
+   ${COPY} ${DARTROOT}/models/POP/shell_scripts/${cesmtagmajor}/CESM_DART_config .
+endif
+
+if (   -e CESM_DART_config ) then
+   sed -e "s#BOGUS_DART_ROOT_STRING#$DARTROOT#" < CESM_DART_config >! temp.$$
+   ${MOVE} temp.$$ ${CASEROOT}/CESM_DART_config
+   chmod 755       ${CASEROOT}/CESM_DART_config
+else
+   echo "WARNING: the script to configure for data assimilation is not available."
+   echo "         CESM_DART_config should be present locally or in"
+   echo "         ${DARTROOT}/models/POP/shell_scripts/${cesmtagmajor}/"
+   echo "         You can stage this script later, but you must manually edit it"
+   echo "         to reflect the location of the DART code tree."
+endif
+
+# ==============================================================================
+# Configuring the case.
+# ==============================================================================
```

Comment:
Before configuring the case (which changes the env*xml files), we have made copies of the env*xml files with their original contents - for comparison, mostly.  Is this step not wanted? Is there some other way to compare ... some other copy to compare against?

# Save a copy for debug purposes
foreach FILE ( *xml )
   if ( ! -e        ${FILE}.original ) then
      ${COPY} $FILE ${FILE}.original
   endif
end